app/models/vm_or_template/operations/configuration.rb

```diff
@@ -74,8 +74,10 @@ def disconnect_floppies
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
     if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
-      raise _("Data Store does not exist, unable to add disk") unless datastore
+      datastore = ext_management_system.hosts.collect do |h|
+        h.writable_storages.find_by(:name => options[:datastore])
+      end.uniq.compact.first
+      raise _("Data Store does not exist or cannot be accessed, unable to add disk") unless datastore
```

Comment:
We should use `Datastore` to match usage through the rest of the product.
https://github.com/ManageIQ/manageiq/blob/master/locale/en/manageiq.po#L801

Only place it differs is with the previous error string from the method: https://github.com/ManageIQ/manageiq/blob/master/locale/en/manageiq.po#L7491